javascript/packages/fury/lib/meta/MetaString.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+export class MetaString {
+  // Encode function that handles all Unicode characters
+  static encode(str: string): Uint8Array {
+    const encoder = new TextEncoder();
```

Comment:
You can not replace the `encoding algorithm` which was implemented on the previous PRs to TextEncoder. TextEncoder is used for encoding the string to a utf8 buffer. The encoding algorithm in our protocol (https://fury.apache.org/docs/specification/fury_xlang_serialization_spec#encoding-algorithms) is used for compressing the string to a minimum byte size.  Please revert this file to the last pr

Reply:
The TextEncoder/Decoder native functions have been removed.

Reply:
> The TextEncoder/Decoder native functions have been removed.
The main issue is not whether to use TextEncoder or not, but that you need to implement our protocol. 
Neither `TextEncoder` nor `atob` can meet our requirements because our protocol is private and not a simple conversion from char to numbers. ()
The code you submitted in (https://github.com/apache/fury/pull/1675/commits/e5a9c3acc120e73e403b6860f81c630480141094) is right.